### PR TITLE
ENH: Updated Spatial Object tex to match ITKv5

### PR DESCRIPTION
**Pull request by aylward**
This pull request is paired with a pull request for ITKSoftwareGuide (pull request has the same name) to update the tex/documentation of SpatialObjects to match the implementation in ITKv5

### CODE REVIEW

File: Examples/SpatialObjects/SpatialObjectHierarchy.cxx [hunk 1]
```diff
@@ -137,11 +150,17 @@ int main( int , char *[] )
 //
 // The \code{Clear()} method erases all the information regarding the object
 // as well as the data. This method is usually overloaded by
-// derived classes.
+// derived classes.  Note that the Parent-Child relationships of the object
+// are NOT reset when \code{Clear()} is called; however, the object-to-parent
+// transform is reset to Identity.  As a result, \code{Update()} should be
+// called before the object is re-used, to re-compute convenience member
+// variables and values.  To remove the children of a node, use the
+// \code{RemoveAllChildre()} function.
```
Comment (dzenanz) [inline Examples/SpatialObjects/SpatialObjectHierarchy.cxx]: RemoveAllChildre -> RemoveAllChildre**n**

File: Examples/SpatialObjects/SpatialObjectTransforms.cxx [hunk 2]
```diff
@@ -22,53 +22,65 @@
...
 //
 // Several member functions and variables are available to every SpatialObject
 // so that they can readily access the WorldSpace in which they exist:
 //
-// * ProtectedComputeObjectToWorldTransform: Composes its ObjectToParentTransform
-// with its parent's cached ObjectToObjectToWorldTransform, to determine the
-// transform from the object's ObjectSpace to WorldSpace.   This transform is
+// \begin{description}
+//
+// \item[ProtectedComputeObjectToWorldTransform()]: This function is called
```
Comment (dzenanz) [inline Examples/SpatialObjects/SpatialObjectTransforms.cxx]: ProtectedComputeObjectToWorldTransform - wasn't this method renamed?

File: Examples/SpatialObjects/SpatialObjectTransforms.cxx [hunk 3]
```diff
@@ -22,53 +22,65 @@
...
-//
-// * GetObjectToWorldTransform: Returns the cached ObjectToWorldTransform.
-// It is the user's responsibility to call ProtectedComputeObjectToWorldTransform when
-// necessary, prior to calling GetObjectToWorldTransform, otherwise the
+// update their cached \code{ObjectToWorldTransform}.   This function should be
+// called on the top level object (via \code{Update()}) once all children
+// object's
+// \code{ObjectToParentTransforms} have been set.   This function should
+// be called
+// on children objects when their \code{ObjectToParentTransforms} have been
```
Comment (dzenanz) [inline Examples/SpatialObjects/SpatialObjectTransforms.cxx]: \code{ObjectToParentTransforms} -> \code{ObjectToParentTransform}s
Plural s outside of `\code`

File: Examples/SpatialObjects/SpatialObjectTransforms.cxx [hunk 4]
```diff
@@ -22,53 +22,65 @@
...
+// changed.
+//
+// \item[GetObjectToWorldTransform()]: Returns the cached
+// \code{ObjectToWorldTransform}.
+// It is the user's responsibility to call \code{Update()} (and thereby
+// \code{ProtectedComputeObjectToWorldTransform()}) when
```
Comment (dzenanz) [inline Examples/SpatialObjects/SpatialObjectTransforms.cxx]: I just looked in the code, I guess `ProtectedComputeObjectToWorldTransform` survived.

File: Examples/SpatialObjects/SpatialObjectHierarchy.cxx (hunk 1 above)
Comment (aylward) [reply Examples/SpatialObjects/SpatialObjectHierarchy.cxx]: Fixed.

File: Examples/SpatialObjects/SpatialObjectTransforms.cxx (hunk 2 above)
Comment (aylward) [reply Examples/SpatialObjects/SpatialObjectTransforms.cxx]: I wish - but I decided not to fight that battle :)    It is named "ProtectedComputeObjectToWorldTransform()"...for now...

File: Examples/SpatialObjects/SpatialObjectTransforms.cxx (hunk 3 above)
Comment (aylward) [reply Examples/SpatialObjects/SpatialObjectTransforms.cxx]: Fixed here and elsewhere.

File: Examples/SpatialObjects/SpatialObjectTransforms.cxx (hunk 4 above)
Comment (aylward) [reply Examples/SpatialObjects/SpatialObjectTransforms.cxx]: Agree.

**Comment by thewtex** [comment]
Also pushed to the `release` branch.

**Pull request merged**